tools/test.py

```diff
@@ -138,6 +138,14 @@ def RunSingle(self, parallel, thread_id):
       try:
         start = datetime.now()
         output = case.Run()
+        # SmartOS has a bug that causes unexpected ECONNREFUSED errors.
+        # See https://smartos.org/bugview/OS-2767
+        # If ECONNREFUSED on SmartOS, retry the test one time.
+        if (output.UnexpectedOutput() and
+          sys.platform == 'sunos5' and 
+          'ECONNREFUSED' in output.output.stderr):
+            sys.stderr.write('ECONNREFUSED encountered on SmartOS; retrying test.')
```

Comment:
I created a new branch and moved the `write()` statement outside of the `if` so that it triggered every time. It didn't seem to mess up TAP. https://ci.nodejs.org/job/node-test-commit-smartos/387/nodes=smartos14-32/console

So, we could leave it in if the info is helpful. (I'll have to add a `\n` to the print string, but that's easy enough of course.)